doc/design-component.md

```diff
@@ -0,0 +1,69 @@
+Component Loader Design
+=======================
+
+# Background
+MicroShift is designed to provide a turnkey solution that can readily serve the target workloads 
+with little intervention, probably in an airgapped environment. This requires MicroShift to
+be able to provides the most, if not all, infrastructure components during bootstrapping. 
+
+In its current form, MicroShift has an predefined asset repo and a sequence of 
```

Comment:
"... asset repo that gets compiled into the MicroShift binary and loaded into the running cluster." ?